### PR TITLE
fix: allow resetting password if using Fido keys

### PR DESCRIPTION
**Pull request by AntoineAugusti**
If you had a Fido key associated with your account, it was not possible to reset your password. It was not updated in the database. This PR fixes that.

Trello: https://trello.com/c/qTsarEod/596-yubikey-cant-reset-password

### CODE REVIEW

File: app/main/views/user_profile.py [hunk 1]
```diff
@@ -280,14 +280,17 @@ def user_profile_validate_security_keys():
...
-    if(session.get('user_details')):
+    if session.get('user_details'):
         user_id = session['user_details']['id']
     else:
         user_id = current_user.id
 
     resp = user_api_client.validate_security_keys(user_id, payload)
     user = User.from_id(user_id)
 
+    if 'password' in session.get('user_details', {}):
```
Comment (AntoineAugusti) [inline app/main/views/user_profile.py:291]: For Fido keys, this is the endpoint that's called https://github.com/cds-snc/notification-admin/blob/d7d7b59c8082d2017aebcf09a2b4f2c34d64dbb5/app/assets/javascripts/fido2.js#L20-L37

Other methods go here https://github.com/cds-snc/notification-admin/blob/d7d7b59c8082d2017aebcf09a2b4f2c34d64dbb5/app/main/views/two_factor.py#L77-L92

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/a4aeba0334fa980af227af8979befe9c141822ac

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/9c743bb80bc5c501e01950bca0ba3ec3ca4dfd82

File: app/main/views/user_profile.py [hunk 2]
```diff
@@ -280,14 +280,17 @@ def user_profile_validate_security_keys():
...
+        user.update_password(session['user_details']['password'])
```
Comment (QuanNguyen-CDS) [inline app/main/views/user_profile.py:292]: do we have a password policy?
i.e. should the password be not empty, a range of lengths, a max length, contain specific characters, must contain a character type, etc.

Comment (AntoineAugusti) [reply app/main/views/user_profile.py:292]: We do! At this point the password as already been validated, it has been set in the new password endpoint https://github.com/cds-snc/notification-admin/blob/97c59815120e6ae72c60c10b06aba8c69c3bc294/app/main/views/new_password.py#L23 and the password field enforces that the password is at least 8 characters, not on a blocklist and not on have I been pwned https://github.com/cds-snc/notification-admin/blob/97c59815120e6ae72c60c10b06aba8c69c3bc294/app/main/forms.py#L189

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/2c57e36cfb95013ef647ede002fd0fded3c70090

**Review by QuanNguyen-CDS — APPROVED** [review]

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/c415adb07610ec2de34f7154801f91a85cc94451

**Comment by maxneuvians** [comment]
https://a11y-tools-query.herokuapp.com/revisions/cds-snc%2Fnotification/af89c0453256907f6e1af1555fd0dc4715d1db25

**Pull request merged**